ui/pages/settings/password-reset.js

```diff
@@ -0,0 +1,108 @@
+import useSWR from 'swr'
+import { useState } from 'react'
+import { useRouter } from 'next/router'
+import Head from 'next/head'
+import Link from 'next/link'
+
+import Fullscreen from '../../components/layouts/fullscreen'
+import ErrorMessage from '../../components/error-message'
+
+export default function PasswordReset () {
+  const router = useRouter()
+
+  const { data: auth } = useSWR('/api/users/self')
+
+  const [password, setPassword] = useState('')
+  const [confirmPassword, setConfirmPassword] = useState('')
+  const [error, setError] = useState('')
+
+  async function onSubmit (e) {
+    e.preventDefault()
+
+    if (password !== confirmPassword) {
+      setError('password does not match')
+      return false
+    }
+
+    setError('')
+
+    try {
+      const rest = await fetch(`/api/users/${auth?.id}`, {
+        method: 'PUT',
+        body: JSON.stringify({
+          id: auth?.id,
```

Comment:
For this one we should do something like
```
{
...auth,
password
}
```

To capture any future fields since PUT requires all fields to be present